Permissions

topless opened the pull request.

topless committed:
Working on permissions.

topless committed:
Editor module permissions

- Configurable permissions.
- Default permission for example app allow all.
- Introduced utils with relevant permission decorators.

Co-authored-by: Nicola Tarocco <nicola.tarocco@cern.ch>

topless committed:
Fixed pydocstyle lint errors.

topless committed:
Restored accidental removal.

topless committed:
Importing permission from app not config directly.

topless committed:
Editor module docs and tests

- removed fixtures script
- adjusted setup/teardown scripts
- initialised third party modules in test suit
- fixed test_view
- removed test_example_app_role_admin
- removed documentation for config could not resolve permissions import.

topless commented:
Hey @ntarocco and @slint can you have a look on this? By the way you can't hurt my feelings. Cheers!

ntarocco commented on invenio_record_editor/utils.py:
```diff
@@ -0,0 +1,41 @@
+# Invenio-Record-Editor is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio module serving a generic record editor."""
```
Better comment?

ntarocco commented on invenio_record_editor/views.py:
```diff
@@ -7,28 +7,27 @@
+@need_editor_permissions('editor-view')
 def index():
-    """Render a basic view."""
+    """Render a basic view, with dummy permission editor-view."""
```
Well, it is not dummy normally :)

`Render the view to display the record editor` or similar?

ntarocco commented on invenio_record_editor/views.py:
```diff
@@ -7,28 +7,27 @@
 
 """Invenio module serving a generic record editor."""
```
Comment?

ntarocco commented on setup.py:
```diff
@@ -37,11 +37,15 @@
 install_requires = [
+    'Flask-BabelEx>=0.9.2',
     'invenio-assets>=1.0.0',
+    'invenio-db>=1.0.2',
```
@Alex (cannot tag you), here we should require db with the optional sqlite, etc... actually, db is not required for this module, right?

slint replied:
If it's only used in tests/example-app it should go in `tests_require` (or `example/requirements.txt`) like e.g. `invenio-records-rest` does: https://github.com/inveniosoftware/invenio-records-rest/blob/788443596aad1d86a8a06b17f7625692a30e6092/setup.py#L23

ntarocco replied:
@topless can u fix this?

topless replied:
@ntarocco at the most recent commit I address the aforementioned issues. Tests and example app are running and the only dependency left is `accounts` for the `jwt` token.

ntarocco reviewed (commented):
In general, check the doc strings around.
Also, build the documentation, fire up the index.html and see how it looks like, it might be that you have to improve it.

topless committed:
Loading app config in blueprint

Function to create the blueprint so we can register it after the wrapper application has initialised so we can use the overriding configuration.

Co-authored-by: Zacharias Zacharodimos <zacharias.zacharodimos@cern.ch>

slint commented on setup.py:
```diff
@@ -37,11 +37,15 @@
+    'Flask-BabelEx>=0.9.2',
     'invenio-assets>=1.0.0',
+    'invenio-db>=1.0.2',
+    'invenio-access>=1.0.0',
```
I'm about to say that you might actually not need `invenio-access` at all... The `AllowAll` permission is not depending on anything from `invenio_access`

topless replied:
Exactly my thought, now that it works end to end, I 'll start killing dependencies.

slint reviewed (approved):
LGTM :shipit:

topless committed:
Refactored permissions, added documentation

- Removed utils and simplified permission class to pass sphinx
- Edited some documentation

topless committed:
Secure JWT token, and bumped editors version

topless committed:
Fixed dependencies for example app

topless committed:
Removed dependencies, added docstrings, integrated invenio-accounts

ntarocco commented:
You can change this PR to go to https://github.com/inveniosoftware/invenio-records-editor

topless committed:
Fixed tests and example app, moved dependencies where they belong

topless committed:
Changes to reflect removal of lazy loading for editor.

topless committed:
Beautification commit.

- Remove style override, everything comes bundled from the angular app.
- Removed invenio style, not extending default template any longer but start clean.
